poc_mobile_verifier/src/heartbeats.rs

```diff
@@ -193,11 +186,21 @@ impl Heartbeat {
 
         Ok(sqlx::query_as::<_, HeartbeatSaveResult>(
             r#"
-            insert into heartbeats (hotspot_key, cbsd_id, reward_weight, timestamp)
-            values ($1, $2, $3, $4)
-            on conflict (cbsd_id) do update set
-            hotspot_key = EXCLUDED.hotspot_key, reward_weight = EXCLUDED.reward_weight, timestamp = EXCLUDED.timestamp
-            returning (xmax = 0) as inserted;
+            INSERT INTO heartbeats (hotspot_key, cbsd_id, reward_weight, timestamps)
+            VALUES ($1, $2, $3, ARRAY[$4])
+            ON CONFLICT (cbsd_id) DO UPDATE SET
+            timestamps = CASE WHEN heartbeats.hotspot_key = EXCLUDED.hotspot_key THEN
```

Comment:
if heartbeats are against the radio and not the hotspot why are we gauging these based on whether or not the hotspot key is the same from one heartbeat to the next? shouldn't swapping a radio between different hotspots only affect the speedtest records?

Reply:
We have to reset the number of heartbeats seen because of gamification concerns. Otherwise, a person could swap a radio between two hotspots every thirty minutes and get double the rewards

Reply:
yep, makes sense